inspirehep/modules/records/receivers.py

```diff
@@ -206,3 +215,45 @@ def enhance_after_index(sender, json, record, *args, **kwargs):
 
     elif is_data(json):
         populate_citations_count(record=record, json=json)
+
+
+@after_record_insert.connect
+@after_record_update.connect
+def after_update(sender, *args, **kwargs):
+    json = kwargs.get('record')
+
+    if json.get('deleted'):
+        # in this way the references diff will be the list of references itself
+        record = InspireRecord(json)
+    else:
+        pid_type = get_pid_type_from_schema(json['$schema'])
+        pid_value = json['control_number']
+        try:
+            rec_uuid = PersistentIdentifier.get(pid_type, pid_value).object_uuid
+            record = InspireRecord.get_record(rec_uuid)
+        except NoResultFound:
+            # deleted record
+            record = InspireRecord(json)
+
+    modified_references = record.get_modified_references()
+
+    pids = [
+        str(get_recid_from_ref(ref['record'])) for ref in modified_references
```

Comment:
this should really use `get_pid_from_record_uri` so it also works with citations to data: https://github.com/inspirehep/inspire-next/blob/b4e9b80149dbf7d6a922742753fd9b71e845a856/inspirehep/modules/records/utils.py#L84-L90.

Reply:
done